drv/fpga-app-api/src/power_rail.rs

```diff
@@ -0,0 +1,92 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use drv_fpga_api::FpgaError;
+use userlib::FromPrimitive;
+use zerocopy::{AsBytes, FromBytes};
+
+/// This module implements state primitives matching the generic power
+/// rail/voltage regulator as implemented in
+/// https://github.com/oxidecomputer/quartz/blob/main/hdl/power_rail.rdl,
+/// https://github.com/oxidecomputer/quartz/blob/main/hdl/PowerRail.bsv.
+
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    Default,
+    Eq,
+    PartialEq,
+    FromBytes,
+    FromPrimitive,
+    AsBytes,
+)]
+#[repr(C)]
+pub struct PowerRailState(u8);
```

Comment:
Since this doesn't actually validate that the `u8` is valid, I'd suggest calling it `RawPowerRailState` instead.